pkg/network/protocols/http/etw_http_service.go

```diff
@@ -977,7 +986,10 @@ func httpCallbackOnHTTPRequestTraceTaskSrvdFrmCache(eventInfo *etw.DDEventRecord
 
 		// <<<MORE ETW HttpService DETAILS>>>
 		httpConnLink.http.SiteID = cacheEntry.http.SiteID
-		httpConnLink.http.SiteName = iisConfig.GetSiteNameFromID(cacheEntry.http.SiteID)
+		cfg := iisConfig.Load()
+		if cfg != nil {
+			httpConnLink.http.SiteName = cfg.GetSiteNameFromID(cacheEntry.http.SiteID)
```

Comment:
Curious, why does the callback need to load the IIS config when fetching values from a cache?

Reply:
Good question, which I'm not totally sure.
Note this behavior is preexisting; the change here is to prevent dereferencing a `nil` pointer.

Reply:
I do not think it was me. I think, if remember my 2 years old implementation that pool name came with ETW payload as well as SiteID but not SiteName (just above `httpConnLink.http.SiteID = userData.GetUint32(16)`), I think. I think it was you Derek who added correlating SiteID to the Site name a while ago by getting it from a map loaded from config

Reply:
Ah, I think Alex, got fooled by load name. It is not actually load in the sense initialization per se but atomic copying from a variable. Are we on the same page?

Reply:
resolved (I think) in slack